internal/pkg/action/operator_uninstall.go

```diff
@@ -86,45 +87,39 @@ func (u *OperatorUninstall) Run(ctx context.Context) error {
 		return err
 	}
 
-	if csv != nil {
-		// Ensure CustomResourceDefinitions are deleted next, so that the operator
-		// has a chance to handle CRs that have finalizers.
-		if u.DeleteCRDs {
-			crds := getCRDs(csv)
-			if err := u.deleteObjects(ctx, crds...); err != nil {
-				return err
-			}
-		}
-
-		// OLM puts an ownerref on every namespaced resource to the CSV,
-		// and an owner label on every cluster scoped resource. When CSV is deleted
-		// kube and olm gc will remove all the referenced resources.
-		if err := u.deleteObjects(ctx, csv); err != nil {
-			return err
-		}
+	operands, err := u.operandLister.Run(ctx, u.Package)
+	if err != nil {
+		return fmt.Errorf("listing operands associated with operator %s: %s", u.Package, err)
 	}
+	u.Logf("found %d operands associated with operator %s", len(operands.Items), u.Package)
+	u.Logf("operand deletion strategy: %s", u.OperandStrategy.String())
 
-	if u.DeleteOperatorGroups {
-		subs := v1alpha1.SubscriptionList{}
-		if err := u.config.Client.List(ctx, &subs, client.InNamespace(u.config.Namespace)); err != nil {
-			return fmt.Errorf("list subscriptions: %v", err)
+	switch u.OperandStrategy {
+	case operand.Cancel:
+		if len(operands.Items) > 0 {
+			return fmt.Errorf("deleting an operator with operands on-cluster requires a operand deletion strategy to be set")
 		}
-		// If there are no subscriptions left, delete the operator group(s).
-		if len(subs.Items) == 0 {
-			ogs := v1.OperatorGroupList{}
-			if err := u.config.Client.List(ctx, &ogs, client.InNamespace(u.config.Namespace)); err != nil {
-				return fmt.Errorf("list operatorgroups: %v", err)
-			}
-			for _, og := range ogs.Items {
-				og := og
-				if len(u.DeleteOperatorGroupNames) == 0 || contains(u.DeleteOperatorGroupNames, og.GetName()) {
-					if err := u.deleteObjects(ctx, &og); err != nil {
-						return err
-					}
-				}
+	case operand.Ignore:
+		u.Logf("ignoring %d operands on-cluster managed by operator %s", len(operands.Items), u.Package)
+	case operand.Delete:
+		for _, op := range operands.Items {
+			// TODO parallelize?
+			op := op
+			if err := u.deleteObjects(ctx, &op); err != nil {
+				return err
 			}
 		}
+	default:
+		return fmt.Errorf("unknown operand deletion strategy provided: %s", u.OperandStrategy.String())
```

Comment:
Nit: keep it short and sweet
Suggestion: use `%q` when printing strings that may be empty or are user input. `%s` with empty strings is a pretty crummy UX
```suggestion
		return fmt.Errorf("unknown operand deletion strategy %q", u.OperandStrategy)
```